diff -Naur lpeg-0.12.orig/lptree.c lpeg-0_12/lptree.c
--- lpeg-0.12.orig/lptree.c	Fri Apr 12 09:31:19 2013
+++ lpeg-0_12/lptree.c	Fri Aug  2 10:15:43 2013
@@ -1217,8 +1217,7 @@
 };
 
 
-int luaopen_lpeg (lua_State *L);
-int luaopen_lpeg (lua_State *L) {
+LUALIB_API int luaopen_lpeg (lua_State *L) {
   luaL_newmetatable(L, PATTERN_T);
   lua_pushnumber(L, MAXBACK);  /* initialize maximum backtracking */
   lua_setfield(L, LUA_REGISTRYINDEX, MAXSTACKIDX);
diff -Naur lpeg-0.12.orig/CMakeLists.txt lpeg-0.12/CMakeLists.txt
--- lpeg-0.12.orig/CMakeLists.txt	Wed Dec 31 16:00:00 1969
+++ lpeg-0.12.orig/CMakeLists.txt	Fri Aug  2 09:26:42 2013
@@ -0,0 +1,25 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+project(lpeg C)
+
+set(LPEG_SOURCE
+lpcap.c
+lpcode.c
+lptree.c
+lpvm.c
+)
+
+include_directories("${EP_BASE}/include/luajit-2.0")
+if (WIN32)
+    add_library(lpeg SHARED ${LPEG_SOURCE})
+    add_definitions(-DLUA_BUILD_AS_DLL -DLUA_CORE)
+    target_link_libraries(lpeg "${EP_BASE}/lib/lua51.lib")
+else()
+    add_library(lpeg STATIC ${LPEG_SOURCE})
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -Wall -fPIC")
+endif()
+
+install(TARGETS lpeg DESTINATION lib)
